File: 4_largest_palindrome/4_largest_palindrome_product.c
// Problem is here: https://projecteuler.net/problem=4
// Code that finds the largest palindrome that is a
// product of two 3 digit numbers
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

/*
  Function that takes in a number and checks if it's palindrome, returns 0 if it isn't
*/
int CheckIfPalindrome(int number)
{
  char string_form[64];
  sprintf(string_form,"%d",number);
  int first = 0;
  int last = strlen(string_form) - 1;

  while (first < last)
  {
    if (string_form[first] != string_form[last])
    {
      return 0;
    }
    first++;
    last--;
  }
}

int main()
{
  int number;
  int largest_palindrome = 0;

  int counter1;
  int counter2;
  for (counter1=999;counter1>1;counter1--)
  {
    for (counter2=999;counter2>1;counter2--)
    {
      number = counter1*counter2;
      if ((CheckIfPalindrome(number) != 0) && (number > largest_palindrome))
      {
        largest_palindrome = number;

      }
    }
  }
  printf("\nLargest palindrome: %d\n",largest_palindrome);

}

File: 4_largest_palindrome/4_largest_palindrome_product.py
# Problem is here: https://projecteuler.net/problem=4
# Code that finds the largest palindrome that is a
# product of two 3 digit numbers

#  Function that takes in a number and checks if it's a palindrome
def CheckIfPalindrome(number):
    string_form = str(number)
    for i in range(int(len(string_form)/2)):    # int(len(string_form)) will give a rounded down version of half the length
        if string_form[i] != string_form[-(i+1)]:
            return 0
    return 1

# Looping through multiplying two 3 digit numbers
largest_palindrome = 0
for i in range(999):
    for j in range(999):
        check = 1
        number = i * j
        check = CheckIfPalindrome(number)
        if check == 1 and number > largest_palindrome:
            largest_palindrome = number
            print(number)

print("The largest palindrome is: " + str(largest_palindrome))
